R/TuneControlRandom.R

```diff
@@ -3,15 +3,19 @@
 #'   Default is 100.
 #' @export
 #' @rdname TuneControl
-makeTuneControlRandom = function(same.resampling.instance = TRUE, maxit = 100L, tune.threshold = FALSE,
+makeTuneControlRandom = function(same.resampling.instance = TRUE, maxit = NULL, tune.threshold = FALSE,
   tune.threshold.args = list(), log.fun = NULL, final.dw.perc = NULL, budget = NULL) {
 
-  if (is.null(budget))
+  if (is.null(budget)) {
+    if (is.null(maxit))
+      maxit = 100L
     budget = maxit
-  else if (is.null(maxit))
-    maxit = budget
-  else if (budget != maxit)
-    stopf("The parameters budget (%i) and maxit (%i) differ.", budget, maxit)
```

Comment:
[GenSA](https://github.com/mlr-org/mlr/blob/d7005d1408d8379258b9de8868cf98cf409ef555/R/TuneControlGenSA.R#L8-L21) and [IRace](https://github.com/mlr-org/mlr/blob/d7005d1408d8379258b9de8868cf98cf409ef555/R/TuneControlIrace.R#L21-L31) do it pretty much like this. What do you mean 'without having to change the defaults of makeTuneControlRandom'?